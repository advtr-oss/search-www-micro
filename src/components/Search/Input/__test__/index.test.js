/**
 * @jest-environment jsdom
 */

import React from 'react'
import { render, screen, act, fireEvent, waitFor } from '@testing-library/react'

import { Input } from '../index'

describe('Input', function () {
  it('should display the base input', function () {
    const { container } = render(<Input />)
    expect(container.firstChild).toMatchSnapshot()
  })


  it('should hide icon', async function () {
    const { getByLabelText } = render(<Input hide />)

    const icn = getByLabelText('input-icon')
    expect(icn).not.toBeVisible()
  })

  it('should display a disabled base input', function () {
    const { container } = render(<Input disabled />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should clear button click', async function () {
    // Just to test `handleClick` was called
    const handleClear = jest.fn()

    const { getByLabelText } = render(<Input onClear={handleClear} hide={false} />)

    act(() => {
      // Only works on the SVG
      const btn = getByLabelText('input-icon')
      fireEvent.click(btn)
    })

    await waitFor(() => expect(handleClear).toBeCalledTimes(1))
  })

  it('should handle text input', async function () {
    // Just to test `handleInput` was called
    const handleInput = jest.fn()

    render(<Input onInput={handleInput} value={''} />)
    expect(screen.getByLabelText('Search autocomplete').value).toBe('')

    await act(() => {
      fireEvent.input(screen.getByLabelText('Search autocomplete'), {
        target: { value: 'Hello World' }
      })
    })

    expect(screen.getByLabelText('Search autocomplete').value).toBe('Hello World')
    expect(handleInput).toBeCalledTimes(1)
  });
})
