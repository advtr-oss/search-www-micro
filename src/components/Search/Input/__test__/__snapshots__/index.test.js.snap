// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should display a disabled base input 1`] = `
.c2 {
  width: 100%;
  border-style: none;
}

.c2:focus {
  outline: none;
}

.c2:focus {
  outline: none;
}

.c2:focus {
  outline: none;
}

.c2:focus {
  outline: none;
}

.c0 {
  position: relative;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: none;
  padding: 0 8px;
}

.c0 .c1 {
  padding: 0 8px 0 8px;
  background-color: inherit;
  line-height: 40px;
}

.c3 {
  display: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  padding-left: 8px;
  background-color: inherit;
  border-style: none;
  border-radius: 10px;
}

.c3:disabled {
  cursor: default;
}

.c3:disabled > svg {
  cursor: default;
}

<div
  class=""
>
  <div
    class="c0"
  >
    <input
      aria-label="Search autocomplete"
      autocomplete="off"
      class="c1 c2 input--disabled input--empty"
      disabled=""
      placeholder="Input..."
      type="text"
      value=""
    />
    <button
      class="c3"
      disabled=""
    >
      <svg
        aria-label="input-icon"
        disabled=""
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <line
          x1="18"
          x2="6"
          y1="6"
          y2="18"
        />
        <line
          x1="6"
          x2="18"
          y1="6"
          y2="18"
        />
      </svg>
    </button>
  </div>
</div>
`;

exports[`Input should display the base input 1`] = `
.c2 {
  width: 100%;
  border-style: none;
}

.c2:focus {
  outline: none;
}

.c2:focus {
  outline: none;
}

.c2:focus {
  outline: none;
}

.c2:focus {
  outline: none;
}

.c0 {
  position: relative;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: none;
  padding: 0 8px;
}

.c0 .c1 {
  padding: 0 8px 0 8px;
  background-color: inherit;
  line-height: 40px;
}

.c3 {
  display: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  padding-left: 8px;
  background-color: inherit;
  border-style: none;
  border-radius: 10px;
}

.c3:disabled {
  cursor: default;
}

.c3:disabled > svg {
  cursor: default;
}

<div
  class=""
>
  <div
    class="c0"
  >
    <input
      aria-label="Search autocomplete"
      autocomplete="off"
      class="c1 c2 input--empty"
      placeholder="Input..."
      type="text"
      value=""
    />
    <button
      class="c3"
    >
      <svg
        aria-label="input-icon"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <line
          x1="18"
          x2="6"
          y1="6"
          y2="18"
        />
        <line
          x1="6"
          x2="18"
          y1="6"
          y2="18"
        />
      </svg>
    </button>
  </div>
</div>
`;
